CRM/Core/BAO/Note.php

```diff
@@ -270,53 +270,53 @@ public static function &getValues(&$params, &$values, $numNotes = self::MAX_NOTE
     return $notes;
   }
 
+  /**
+   * Event fired prior to modifying a Note.
+   * @param \Civi\Core\Event\PreEvent $event
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    if ($event->action === 'delete' && $event->id) {
+      // When deleting a note, also delete child notes
+      // This causes recursion as this hook is called again while deleting child notes,
+      // So the children of children, etc. will also be deleted.
+      foreach (self::getDescendentIds($event->id) as $child) {
+        self::deleteRecord(['id' => $child]);
+      }
+    }
+  }
+
   /**
    * Delete the notes.
    *
    * @param int $id
    *   Note id.
    * @param bool $showStatus
-   *   Do we need to set status or not.
+   *   FIXME: Status messages ought to be set by the form.
    *
-   * @return int|null
-   *   no of deleted notes on success, null otherwise
+   * @deprecated
+   * @return int
    */
   public static function del($id, $showStatus = TRUE) {
-    $return = NULL;
-    $recent = array($id);
-    $note = new CRM_Core_DAO_Note();
-    $note->id = $id;
-    $note->find();
-    $note->fetch();
-    if ($note->entity_table == 'civicrm_note') {
-      $status = ts('Selected Comment has been deleted successfully.');
-    }
-    else {
-      $status = ts('Selected Note has been deleted successfully.');
+    if ($showStatus) {
+      $note = new CRM_Core_DAO_Note();
+      $note->id = $id;
+      $note->find();
+      $note->fetch();
+      if ($note->entity_table == 'civicrm_note') {
```

Comment:
Honestly I would rip out this status stuff & just fix the 4 places that call this to set the status after it is called. The 'comment' part seems like a really long ago feature idea that never amounted to much